tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/record-variable-definition-stmt.bal

```diff
@@ -743,6 +743,20 @@ function testReadOnlyRecordWithMappingBindingPatternInVarDecl() {
     assertEquality(<ReadOnlyRecord> {x: [1, 2], y: "s1"}, b1);
 }
 
+function recordVariableWithAnonymousType() returns [string, boolean]|error {
+    record {|
+        string name;
+        boolean married;
+    |} {name, married} = check getPersonRecord().ensureType();
```

Comment:
Shall we add a few more tests with nested patterns?

Reply:
I have added a test case for a nested scenario. Could you please confirm whether this is the type of scenario you intended?

Reply:
In https://github.com/ballerina-platform/ballerina-lang/pull/39688/commits/5a0b98caf6c71532c6d80eb619caac6a39588f04#diff-ee5929c4cfdb578fc7e234b7577be5fa4aa66a2687e2b4ffa65fe35cfc62b889R775 we can use another mapping binding pattern for details.